### PR TITLE
DEVOPS-534 Send alerts to Hipchat

### PR DESCRIPTION
**Pull request by avivl**
Send Alerts to a HipChat room

@oryband @doodyparizada do you thing.....

### CODE REVIEW

**Comment by oryband** [comment]
apply comments from https://github.com/rounds/elastalert/commit/cd35d7c8c749bd311b33afd8da1df2337b1f4bfa please

File: elastalert/alerts.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 import json
 import logging
 import subprocess
+from copy import deepcopy
+import requests
```
Comment (oryband) [inline elastalert/alerts.py]: `requests` is 3rd party

File: elastalert/alerts.py [hunk 2]
```diff
@@ -443,3 +445,50 @@ def alert(self, matches):
     def get_info(self):
         return {'type': 'command',
                 'command': ' '.join(self.last_command)}
+
+
+class HipChatAlerter(Alerter):
+
+    """ Send notfication to a HipChat room. """
```
Comment (oryband) [inline elastalert/alerts.py]: no need for spaces in the docstring:
```python
"""Send a notification to a HipChat room."""
```

Comment (oryband) [inline elastalert/alerts.py]: apply to rest of docstrings

File: elastalert/alerts.py [hunk 3]
```diff
@@ -443,3 +445,50 @@ def alert(self, matches):
...
+
+        """ post to hipchat room. """
+
+        _post_params = deepcopy(post_params)
+
+        headers = {'content-type': 'application/json'}
+        uri = '%s%s?auth_token=%s' % (self.base_uri, endpoint, self.auth_token)
+        http = requests.session()
+        response = http.post(uri, data=json.dumps(_post_params), headers=headers, proxies=None)
+        http.close()
```
Comment (oryband) [inline elastalert/alerts.py]: i would send the request in a single statement:
```python
response = http.post(
    uri= '%s...' % (..),
    headers={'...':' ...'},
    data=json.dumps(_post_params),
    proxies=None)  # is this line even necessary?
```

Comment (oryband) [inline elastalert/alerts.py]: also:
```python
class  C(c):
   session = requests.session()
   # ...
   def _post(..):
       session.post(...)  # check the requests docs for how to use sessions, this is a half baked example
```

File: elastalert/alerts.py [hunk 4]
```diff
@@ -443,3 +445,50 @@ def alert(self, matches):
...
+
+        return response
+
+    def alert(self, matches):
+
+        """ send the alerts """
```
Comment (oryband) [inline elastalert/alerts.py]: only classes use a prefix-line space, functions don't
you should `pip isntall flake8 pep257` and run it against this file, it shows syntax and style errors (pep8)

File: elastalert/alerts.py (hunk 1 above)
Comment (oryband) [inline elastalert/alerts.py]: on second thought, we're following upstream's guidelines which don't divide modules, so nm

**Comment by oryband** [comment]
@doodyparizada mentioned something: we shouldn't use jira namings (DEVOPS-XXX) since this is a public repo, which we would pr upstream soon. they wouldn't like weird private branch names like these

**Comment by avivl** [comment]
@doodyparizada since I'm probably not that great with python, could you make this work correctly ?

**Comment by oryband** [comment]
just checkout from this branch to a new branch with the correct name and open a new pr

**Comment by doodyparizada** [comment]
Sure @avivl

**Pull request closed without merging**